src/finch/algebra/operator.py

```diff
@@ -147,3 +171,10 @@ def identity(x):
     "return_type",
     lambda op, x: x,
 )
+
+
+algebra.register_property(minby, "__call__", "return_type", lambda op, a, b: int)
+algebra.register_property(maxby, "__call__", "return_type", lambda op, a, b: int)
+
+algebra.register_property(minby, "__call__", "init_value", lambda op, arg: 0)
```

Comment:
minby can't return 0 as init value, it should return `(-Inf, 0)`.

Reply:
Shouldn't this one be dependent on the datatype? Integer data types don't support infinity. The following code works for me:

```python
from typing import Literal
import numpy as np

def get_dt_bounds(dt: np.dtype, kind: Literal["min", "max"]) -> np.generic:
    dt = np.dtype(dt)
    if kind not in {"min", "max"}:
        raise ValueError(
            "Only `kind='min'` or `kind='max'` are accepted, found "
            f"`{kind=}`."
        )
    if np.isdtype(dt, kind="integral"):
        return dt.type(getattr(np.iinfo(dt), kind))

    if np.isdtype(dt, kind="real floating"):
        return dt.type(np.inf if kind == "max" else -np.inf)

    raise TypeError("Only complex and real floating value types")
```